packages/backend/src/zoekt.ts

```diff
@@ -0,0 +1,35 @@
+import { exec } from "child_process";
+import { AppContext, GitRepository, LocalRepository } from "./types.js";
+
+export const indexGitRepository = async (repo: GitRepository, ctx: AppContext) => {
+    return new Promise<{ stdout: string, stderr: string }>((resolve, reject) => {
+        exec(`zoekt-git-index -index ${ctx.indexPath} ${repo.path}`, (error, stdout, stderr) => {
+            if (error) {
+                reject(error);
+                return;
+            }
+            resolve({
+                stdout,
+                stderr
+            });
+        })
+    });
+}
+
+export const indexLocalRepository = async (repo: LocalRepository, ctx: AppContext, signal?: AbortSignal) => {
+    const excludedDirs = repo.excludedPaths.length > 0 ? repo.excludedPaths : ['.git', '.hg', '.svn'];
```

Comment:
nit: do we always want to ignore .git, .hg, and .svn? If so this may cause these dirs to not be ignored if they provide exclude paths and forget to include those

Reply:
Yea this makes sense, I think it's a fair assumption that these will always be excluded.